thirdeye/thirdeye-frontend/app/utils/anomalies-tree-parser.js

```diff
@@ -0,0 +1,563 @@
+      detectorComponentName.includes("ANOMALY_SUMMARIZE")
+    ) {
+      setGroupsBucket(buckets, child, subEntityName, groupKey);
+    } else {
+      setEntitiesBucket(buckets, child, subEntityName);
+    }
+  }
+
+  return buckets;
+};
+
+/**
+ * Perform drilldown of anomaly grouped by anomaly summarize grouper. This involves generating the breadcrumb information
+ * and component details for the subtree for this anomaly.
+ *
+ * @param {Object} input
+ *   The subtree structure that needs to be parsed
+ *
+ * @return {Object}
+ *   The breadcrumb info and data for populating component comprising of group constituents
+ */
+const parseGroupAnomaly = input => {
+  const output = [];
+  const data = [];
+  const {
+    GROUPS: { DEFAULT_TITLE, COMPONENT_PATH }
+  } = CLASSIFICATIONS;
+  const {
+    id,
+    children,
+    properties: { subEntityName, groupKey }
+  } = input;
+  const breadcrumbInfo = {
+    title: `${subEntityName}/${groupKey}`,
+    id
+  };
+
+  for (const anomaly of children) {
+    const {
+      id,
+      startTime,
+      endTime,
+      metric,
+      dimensions,
+      avgCurrentVal: current,
+      avgBaselineVal: predicted,
+      feedback
+    } = anomaly;
+
+    data.push({
+      id,
+      startTime,
+      endTime,
+      feedback,
+      metric,
+      dimensions,
+      current,
+      predicted
+    });
+  }
+
+  output.push({
+    componentPath: COMPONENT_PATH,
+    title: DEFAULT_TITLE,
+    data
+  });
+
+  return { breadcrumbInfo, output };
+};
+
+/**
+ * Perform drilldown of composite anomaly. This involves generating the breadcrumb information
+ * and component details for the subtree for the composite anomaly
+ *
+ * @param {Object} input
+ *   The subtree structure that needs to be parsed
+ *
+ * @return {Object}
+ *   The breadcrumb info and data for populating child components from input subtree.
+ */
+const parseCompositeAnomaly = input => {
+  const output = [];
+  const buckets = generateBuckets(input);
+  const {
+    METRICS: { KEY: metricKey },
+    GROUPS: { KEY: groupKey }
+  } = CLASSIFICATIONS;
+  const { id, startTime } = input;
+  const breadcrumbInfo = {
+    id,
+    title: getFormattedBreadcrumbTime(startTime)
+  };
+
+  for (const key in buckets) {
+    const entry = buckets[key];
+
+    if (key === metricKey) {
+      output.push(...parseMetricsBucket(entry));
+    } else if (key === groupKey) {
+      output.push(...parseGroupsBucket(entry));
+    } else {
+      output.push(...parseEntitiesBucket(entry));
+    }
+  }
+
+  return { breadcrumbInfo, output };
+};
+
+/**
+ * Perform depth-first-search to retrieve anomaly in the tree
+ *
+ * @param {Number} id
+ *   The id of the anomaly to be searched
+ * @param {Object} input
+ *   The subtree structure comprising the anomaly
+ *
+ * @return {Object}
+ *   The anomaly referenced by id
+ */
+const findAnomaly = (id, input) => {
+  const { id: anomalyId, children } = input;
+
+  if (anomalyId === id) {
+    return input;
+  } else {
```

Comment:
Please keep the code concise. If an if block always executes a return statement, the subsequent else block is unnecessary. 
Please check the Airbnb rule. 16.3 or no-else-return

Reply:
Yeah more than aiming for conciseness I was just trying to keeping it explicit. Anyway will remove the else.